x-pack/plugins/watcher/public/sections/watch_detail/components/watch_detail/watch_history_component.tsx

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import { Moment } from 'moment';
+import React, { Fragment, useEffect, useState } from 'react';
+
+import {
+  activateWatch,
+  deactivateWatch,
+  fetchWatchHistory,
+  fetchWatchHistoryDetail,
+} from '../../../../lib/api';
+
+import { i18n } from '@kbn/i18n';
+
+import { WatchActionStatus } from './watch_action_status';
+
+import {
+  EuiButton,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { DeleteWatchesModal } from 'x-pack/plugins/watcher/public/components/delete_watches_modal';
+
+const WatchHistoryUI = ({
+  intl,
+  watchId,
+  urlService,
+}: {
+  intl: InjectedIntl;
+  watchId: string;
+  urlService: any;
+}) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isActivated, setIsActivated] = useState<boolean>(true);
+  const [history, setWatchHistory] = useState([]);
+  const [isDetailVisible, setIsDetailVisible] = useState<boolean>(true);
+  const [watchesToDelete, setWatchesToDelete] = useState<string[]>([]);
+  const [itemDetail, setItemDetail] = useState<{
+    id?: string;
+    details?: any;
+    watchId?: string;
+    watchStatus?: { actionStatuses?: any };
+  }>({});
+  const [executionDetail, setExecutionDetail] = useState<string>('');
+
+  const kbnUrlService = urlService;
```

Comment:
not sure why this is done

Reply:
addressed in 2b231250011efa8731d2b1d81bdbe518a95b87d3